services/actions/projects.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package actions
+
+import (
+	"context"
+	"time"
+
+	"code.gitea.io/gitea/models/db"
+	issues_model "code.gitea.io/gitea/models/issues"
+	project_model "code.gitea.io/gitea/models/project"
+	"code.gitea.io/gitea/modules/log"
+	"xorm.io/builder"
+)
+
+// Cleanup removes closed issues from project board with close on move flag set to true
+func CleanupProjectIssues(taskCtx context.Context, olderThan time.Duration) error {
```

Comment:
while this function would work with smaller instances, the resource consumption with larger ones can grow significantly with the several `select`s and no limits/pagination.

Could this logic be done with SQL itself, a join on projects where they have closeonmove, with issues that belong to a board that are closed? Then a separate statement to rm the issue from the project? Then in the rm statement(s) you could batch rm the rows from the DB?